rcl_action/src/rcl_action/action_client.c

```diff
@@ -212,8 +214,19 @@ rcl_action_client_init(
 
   // Avoid uninitialized pointers should initialization fail
   *action_client->impl = _rcl_action_get_zero_initialized_client_impl();
+
+  // Remap/Expand the action name
+  char * resolved_action_name = NULL;
+  ret = rcl_node_resolve_name(node, action_name, allocator, false, false, &resolved_action_name);
+  if (RCL_RET_OK != ret) {
+    if (RCL_RET_TOPIC_NAME_INVALID == ret || RCL_RET_SERVICE_NAME_INVALID == ret) {
+      ret = RCL_RET_ACTION_NAME_INVALID;
+    }
+    goto fail;
```

Comment:
rcl_node_resolve_name() can return many types of errors, so the return values should be simplified.
Refer to
https://github.com/ros2/rcl/blob/0723613c89d86f23a4aef70601994b812a6d15a2/rcl/src/rcl/service.c#L124-L139

Reply:
Thank you for the feedback, i have implemented the recommended changes. The only error codes returned in case the name resolution fails are RCL_RET_ACTION_NAME_INVALID, RCL_RET_BAD_ALLOC and RCL_RET_ERROR.